cmd/limactl/shell.go

```diff
@@ -178,7 +180,25 @@ func shellAction(cmd *cobra.Command, args []string) error {
 	} else {
 		shell = shellescape.Quote(shell)
 	}
-	script := fmt.Sprintf("%s ; exec %s --login", changeDirCmd, shell)
+	// Handle environment variable propagation
+	var envPrefix string
+	withEnv, err := cmd.Flags().GetBool("preserve-env")
```

Comment:
I know you changed the option at the last minute from `--with-env` to `--preserve-env`, but the variable name should be matching:

```suggestion
	preserveEnv, err := cmd.Flags().GetBool("preserve-env")
```